nsv13/code/modules/overmap/boarding/boarding.dm

```diff
@@ -28,57 +66,64 @@
-			if("greytide")
-				knpc_types = list(/mob/living/carbon/human/ai_boarder/assistant)
-			if("nanotrasen")
-				knpc_types = list(/mob/living/carbon/human/ai_boarder/ert, /mob/living/carbon/human/ai_boarder/ert/commander, /mob/living/carbon/human/ai_boarder/ert/medic, /mob/living/carbon/human/ai_boarder/ert/engineer)
-
-		var/list/possible_spawns = list()
-		for(var/obj/effect/landmark/patrol_node/node in GLOB.landmarks_list)
-			if(node.get_overmap() == src)
-				possible_spawns += node
-		//Can we establish a drop zone?
-		var/turf/LZ = (possible_spawns?.len) ? get_turf(pick(possible_spawns)) : null
-		if(!LZ)
-			message_admins("KNPC boarder spawn aborted. This ship does not support KNPCs (add some patrol nodes!))")
-			return FALSE
-
-		var/obj/structure/closet/supplypod/centcompod/toLaunch = new /obj/structure/closet/supplypod/syndicate_odst
-		var/shippingLane = GLOB.areas_by_type[/area/centcom/supplypod/supplypod_temp_holding]
-		toLaunch.forceMove(shippingLane)
-		for(var/I = 0; I < amount; I++)
-			var/soldier_type = pick(knpc_types)
-			new soldier_type(toLaunch)
-
-		new /obj/effect/pod_landingzone(LZ, toLaunch)
-		return TRUE
+/** Attempts to spawn knpc boarders for this ship.
+ * Arguments:
+ * * amount: Number of boarders to spawn. Will be automatically choosen, based on SSovermap mode difficulty if !amount.
+ * * faction_selection: String for the faction the boarders belong to. See [/obj/structure/overmap/proc/spawn_boarders] for valid values.
+ * Returns TRUE is successful, FALSE otherwise.
+ */
+/obj/structure/overmap/proc/spawn_knpcs(amount, faction_selection="syndicate")
+	if(!amount)
+		amount = CEILING(1 + (SSovermap_mode.mode.difficulty / 2), 1)
+	var/list/knpc_types = list()
+	switch(faction_selection)
+		if("syndicate")
+			knpc_types = list(/mob/living/carbon/human/ai_boarder/syndicate/pistol, /mob/living/carbon/human/ai_boarder/syndicate/smg, /mob/living/carbon/human/ai_boarder/syndicate/shotgun)
+		if("pirate")
+			knpc_types = list(/mob/living/carbon/human/ai_boarder/space_pirate/pistol, /mob/living/carbon/human/ai_boarder/space_pirate/auto_pistol)
+		if("greytide")
+			knpc_types = list(/mob/living/carbon/human/ai_boarder/assistant)
+		if("nanotrasen")
+			knpc_types = list(/mob/living/carbon/human/ai_boarder/ert, /mob/living/carbon/human/ai_boarder/ert/commander, /mob/living/carbon/human/ai_boarder/ert/medic, /mob/living/carbon/human/ai_boarder/ert/engineer)
+
+	var/list/possible_spawns = list()
+	for(var/obj/effect/landmark/patrol_node/node in GLOB.landmarks_list)
+		if(node.get_overmap() == src)
+			possible_spawns += node
+	//Can we establish a drop zone?
+	var/turf/LZ = (possible_spawns?.len) ? get_turf(pick(possible_spawns)) : null
+	if(!LZ)
+		message_admins("KNPC boarder spawn aborted. This ship does not support KNPCs (add some patrol nodes!))")
+		return FALSE
+
+	var/obj/structure/closet/supplypod/centcompod/toLaunch = new /obj/structure/closet/supplypod/syndicate_odst
```

Comment:
Is there a reason you're casting a regular supply pod to a centcompod, or was this an accident?
```suggestion
	var/obj/structure/closet/supplypod/centcompod/toLaunch = new()
```

Reply:
Not my code, so I'm hesitant to change that.

Reply:
you should probably change it

Reply:
It would make sense to use a syndicate supply pod but I don't know whether that would cause any issues. I'm currently busy with other code and my mental health being a wreck, so changing this is not on the top of my priority list, but I'll try to have a look at it sometime.